fuse_constraints/test/test_marginalize_variables.cpp

```diff
@@ -175,14 +175,15 @@ TEST(MarginalizeVariables, ComputeEliminationOrder)
   auto expected = fuse_constraints::UuidOrdering();
   expected.push_back(x1->uuid());
   expected.push_back(x2->uuid());
-  expected.push_back(x3->uuid());
```

Comment:
I had to change the expected order in this unit test. I believe the unit test was wrong, but I'm not sure how to find out the correct expected elimination order we should get here. Thoughts?

Reply:
I did a quick test using GTSAM's ordering system. The generated order is indeed: x1, x2, l1, x3

Reply:
Cool. I'm just curious. How did you run the test in GTSAM? Is there any example code, or you had to build yours from scratch?

Reply:
I hacked up this unit test to reproduce our example.
https://github.com/borglab/gtsam/blob/develop/gtsam/inference/tests/testOrdering.cpp#L94

A "symbolic" factor graph models the structure of the graph without worrying about the numeric details of the cost functions. And GTSAM uses sequential numbers as variable names (at least at the symbolic level). So x1->0, x2->1, x3->2, l1->3, l2->4.
```
  SymbolicFactorGraph symbolicGraph;
  symbolicGraph.push_factor(0);
  symbolicGraph.push_factor(0, 1);
  symbolicGraph.push_factor(1, 2);
  symbolicGraph.push_factor(0, 3);
  symbolicGraph.push_factor(1, 3);
  symbolicGraph.push_factor(2, 4);
```
The above reproduces the graph structure in our unittest. 

And then we can generate an ordering using COLAMD:
```
  Ordering actual = Ordering::ColamdConstrainedFirst(symbolicGraph, list_of(1)(0));
```
The second parameter is the list of variables to constrain to the front of the ordering.

And the actual ordering passes this test:
```
  Ordering expected = list_of(0)(1)(3)(2)(4);
  EXPECT(assert_equal(expected, actual));
```
which is (x1, x2, l1, x3, l2).

Note that we are actually computing an ordering for just some of the graph...anything that directly connects to x1 or x2. That includes everything except the x3--l2 factor. If we comment out `symbolicGraph.push_factor(2, 4);`, then we reproduce what we are actually computing. And that returns (0, 1, 3, 2) or (x1, x2, l1, x3).

So both situations produce the (x1, x2, l1, x3) ordering, which makes it pretty easy to decide that is correct. 

I contributed to GTSAM for a couple of years while attending GaTech, which is why I know how all this works.

Reply:
Thanks. Awesome!

That looks good and makes sense.